pkg/registry/helpers_test.go

```diff
@@ -113,5 +113,131 @@ var _ = Describe("config cert command tests", func() {
 
 			})
 		})
+
+	})
+
+	Describe("GetRegistryCertOptions with proxy configured", func() {
+		var (
+			tanzuConfigFile    *os.File
+			tanzuConfigFileNG  *os.File
+			caCertDataOpt      string
+			skipCertVerifyOpt  string
+			insecureOpt        string
+			noProxyOpt         string
+			proxyCACertDataB64 string
+			err                error
+		)
+		const (
+			fakeCACertData      = "fake ca cert data"
+			fakeProxyCACertData = "fake proxy ca cert data"
+			testHost            = "test.vmware.com"
+			testHTTPProxyHost   = "192.168.116.1:3128"
+			testHTTPSProxyHost  = "192.168.116.1:3129"
+		)
+
+		BeforeEach(func() {
+			tanzuConfigFile, err = os.CreateTemp("", "config")
+			Expect(err).To(BeNil())
+			os.Setenv("TANZU_CONFIG", tanzuConfigFile.Name())
+
+			tanzuConfigFileNG, err = os.CreateTemp("", "config_ng")
+			Expect(err).To(BeNil())
+			os.Setenv("TANZU_CONFIG_NEXT_GEN", tanzuConfigFileNG.Name())
+
+			os.Setenv("http_proxy", "http://"+testHTTPProxyHost)
+			os.Setenv("https_proxy", "http://"+testHTTPSProxyHost)
+		})
+		AfterEach(func() {
+			os.Unsetenv("TANZU_CONFIG")
+			os.Unsetenv("TANZU_CONFIG_NEXT_GEN")
+			os.RemoveAll(tanzuConfigFile.Name())
+			os.RemoveAll(tanzuConfigFileNG.Name())
+			os.Unsetenv("http_proxy")
+			os.Unsetenv("https_proxy")
+			os.Unsetenv("no_proxy")
+			os.Unsetenv("TANZU_CLI_PROXY_CA_CERT")
+		})
+		JustBeforeEach(func() {
+			caCertDataOptB64 := ""
+			if len(caCertDataOpt) > 0 {
+				caCertDataOptB64 = base64.StdEncoding.EncodeToString([]byte(caCertDataOpt))
+			}
+			cert := &configtypes.Cert{
+				Host:           testHost,
+				CACertData:     caCertDataOptB64,
+				SkipCertVerify: skipCertVerifyOpt,
+				Insecure:       insecureOpt,
+			}
+			err := configlib.SetCert(cert)
+			Expect(err).To(BeNil())
+			os.Setenv("no_proxy", noProxyOpt)
+			os.Setenv("TANZU_CLI_PROXY_CA_CERT", proxyCACertDataB64)
```

Comment:
nit: Might be better to just use constant here for `TANZU_CLI_PROXY_CA_CERT`.

Reply:
updated